src/main/java/me/moodcat/api/RoomAPI.java

```diff
@@ -174,4 +174,28 @@ public NowPlaying getCurrentTime(@PathParam("id") final int roomId) {
         return nowPlaying;
     }
 
+    /**
+     * Process a vote to a song. A vote is either "like" or "dislike".
+     *
+     * @param vote
+     *            The vote.
+     * @return The song object, if the process was succesful.
+     * @throws InvalidVoteException
+     *             Thrown if an invalid vote was cast.
+     */
+    @POST
+    @Path("{id}/vote/{vote}")
+    @Transactional
+    public RoomModel voteSong(@PathParam("id") final int roomId,
+            @PathParam("vote") final String vote) throws InvalidVoteException {
+        final RoomInstance roomInstance = this.backend.getRoomInstance(roomId);
+
+        if ("like".equals(vote) || "dislike".equals(vote)) {
+            roomInstance.addVote(currentUserProvider.get(), Vote.valueOf(vote.toUpperCase()));
+        } else {
+            throw new InvalidVoteException();
```

Comment:
InvalidVoteException extends IllegalArgumentException. We can also use the text of an exception, if you would prefer that?